app/src/main/java/ch/epfl/sdp/game/StartGameController.java

```diff
@@ -70,28 +75,36 @@ static Area initGameArea() {
     /**
      * This method initializes the random enemy generator
      *
-     * @param gameArea     the game area
+     * @param gameArea the game area
+     */
+    void createRandomEnemyGenerator(Area gameArea) {
+        randomEnemyGenerator = new RandomEnemyGenerator(gameArea);
+        randomEnemyGenerator.setMaxEnemies(MAX_ENEMY);
+        randomEnemyGenerator.setMinDistanceFromEnemies(MIN_DIST_FROM_ENEMIES);
+        randomEnemyGenerator.setMinDistanceFromPlayers(MIN_DIST_FROM_PLAYERS);
+    }
+
+    /**
+     * This method generate an enemy
+     *
      * @param enemyManager the enemy manager
      */
-    static void initEnemies(Area gameArea, EnemyManager enemyManager) {
-        // Enemy -------------------------------------------
-        RandomEnemyGenerator randomEnemyGenerator = new RandomEnemyGenerator(gameArea);
-        randomEnemyGenerator.setEnemyCreationTime(1000);
-        randomEnemyGenerator.setMaxEnemies(10);
-        randomEnemyGenerator.setMinDistanceFromEnemies(100);
-        randomEnemyGenerator.setMinDistanceFromPlayers(100);
-        randomEnemyGenerator.generateEnemy(100);
-        Enemy enemy = randomEnemyGenerator.getEnemies().get(0);
-        enemyManager.updateEnemies(enemy);
-        //  -------------------------------------------
+    void generateEnemy(EnemyManager enemyManager) {
+        if(enemyManager.getEnemies().size() < MAX_ENEMY) {
```

Comment:
Isn't this this executed two times? Or was it moved here?

Reply:
This method just calls the method  generateEnemt in randomEnemyGenerator